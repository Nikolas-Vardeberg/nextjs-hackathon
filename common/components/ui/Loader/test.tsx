import Loader from "./index";
import { render } from "@testing-library/react";

describe("Loader", () => {
  it("renders the loader", () => {
    const { getByTestId } = render(<Loader loading>Test</Loader>);
    expect(getByTestId("loader")).toMatchSnapshot();
  });

  it("renders renders the children when not loading", () => {
    const { getByTestId } = render(
      <Loader loading={false}>
        <div data-testid="child">Test</div>
      </Loader>,
    );
    expect(getByTestId("child")).toMatchSnapshot();
  });
});
