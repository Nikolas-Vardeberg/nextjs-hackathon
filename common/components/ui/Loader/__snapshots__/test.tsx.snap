// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders renders the children when not loading 1`] = `
<div
  data-testid="child"
>
  Test
</div>
`;

exports[`Loader renders the loader 1`] = `
<svg
  class="lucide lucide-loader-circle animate-spin h-5 w-5 text-black"
  data-testid="loader"
  fill="none"
  height="24"
  stroke="currentColor"
  stroke-linecap="round"
  stroke-linejoin="round"
  stroke-width="2"
  viewBox="0 0 24 24"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M21 12a9 9 0 1 1-6.219-8.56"
  />
</svg>
`;
